### ISSUE TITLE
multi-molecule input

### ISSUE THREAD

**Issue by sibo** [grimme-lab/xtb#379]
**Is your feature request related to a problem? Please describe.**
Can xtb geometry optimization be run on multi-molecule input XYZ files, returning a multi-molecule output XYZ file? This is convenient for machine learning purposes, where other codes can generate molecular descriptors from multi-molecule input files.

**Describe the solution you'd like**
Ideally each structure of a multi-molecule .xyz file could be optimized with some call like: xtb -forall filename.xyz --opt [--chrg X] [--uhf Y] . The structures would all have to have the same charge and spin multiplicity. There would be some multi-molecule XYZ output file would contain all the optimized structures.

**Describe alternatives you've considered**
I tried using "crest -forall file.xyz -prop reopt". The output crest_property.xyz file has some ghost H atoms added at the origin, and some elements are scrambled. I assume this is because crest tries to align elements and match number of atoms between frames.

**If possible state how you can assist in providing data or code to to implement the feature**
I'm happy to test the code on some multi-molecule .xyz or .sdf files

**Additional context**
Add any other context about the feature request here.

**Comment by awvwgk** [comment]
Depending on the features you want to use in `xtb` this is already possible. For singlepoint calculations multiple instances of `xtb` can coexist in a thread-safe way. The binary doesn't offer this possibility yet, but the C-API and the Python-API allow this already.

If you are looking for a way to batch process multiple (possibly different) structures (in parallel) with the `xtb` binary, I would say this is slightly out-of-scope of the current implementation. The effort to implement a threaded scheduler in `xtb` would be quite significant and probably easier to realize in an external framework.

But I'm all in for making it easier to setup such workflows in external frameworks via the C-API of `xtb`.

**Comment by conradhuebler** [comment]
I have interfaced the C API in https://github.com/conradhuebler/curcuma, thus multi-molecule input (*.xyz) and parallel optimisiation is basicly possible (with GFN2 and GFN1). However, the L-BFGS implementation I used is far less efficient than ANCOPT.

**Comment by sibo** [comment]
@conradhuebler , thank you for sharing your work, which I'll consider using in future studies

Thanks for the suggestion @calvigne. I just tried the mdopt flag on a .xyz file containing two (non-isomeric) structures. It doesn't seem to work. For instance, if the second molecule is the exact same as the first, except one H is substituted with an F, the output file crest_ensemble.xyz has two molecules, both with one F atom. If in the input file the two atoms have differing atom counts, then really strange things happen with crest_ensemble.xyz.

For now, my workaround is to (1) use a python script to split my multi-molecule .xyz into individual .xyz files, and (2) then use bash scripts to automate running xtb on each indiviudal .xyz file and collating the individual xtbopt.xyz into one multimolecule .xyz file. This works fine for occassional use. I did this task more often, I would look into the C-API or Python-API as mentioned by @awvwgk . I still think it would be convenient to implement a simple flag in xtb that would allow for serial optimization of a multi-molecule .xyz file, since not all researchers want to tackle installing and learning APIs, but I will mark this github issue as Closed.
